File: AtividadesAula/aula04/medicao_controller.py
from medicao_model import MedicaoModel
from medicao_view import MedicaoView

class MedicaoController:
    def __init__(self, model: MedicaoModel, view: MedicaoView):
        self.model = model
        self.view = view

    def iniciar(self):
        # Executa e exibe a Rotina 1 (Aleatória)
        rotina1 = self.model.executar_rotina(100000, 100, 100000, True)
        self.view.exibir_tempo_rotina("rotina 1 (Aleatória)", rotina1.tempo_ms)

        # Executa e exibe a Rotina 2 (Sequencial)
        rotina2 = self.model.executar_rotina(100000, 0, 100000, False)
        self.view.exibir_tempo_rotina("rotina 2 (Sequencial)", rotina2.tempo_ms)

File: AtividadesAula/aula04/MedicaoController.java
// CHAMA model.executarRotina() PARA OBTER OS TEMPOS DE EXECUÇÃI E SEM SEGUIDA PARA ESSE RESULTADO PARA view.exibirTempoRotina
public class MedicaoController {
    private MedicaoModel model;
    private MedicaoView view;

    public MedicaoController(MedicaoModel model, MedicaoView view) {
        this.model = model;
        this.view = view;
    }

    public void iniciar() {
        // Executa e exibe a Rotina 1 (Aleatória)
        MedicaoModel.ResultadoMedicao rotina1 = model.executarRotina(100000, 100, 100000, true);
        view.exibirTempoRotina("rotina 1 (Aleatória)", rotina1.tempoMs);

        // Executa e exibe a Rotina 2 (Sequencial)
        MedicaoModel.ResultadoMedicao rotina2 = model.executarRotina(100000, 0, 100000, false);
        view.exibirTempoRotina("rotina 2 (Sequencial)", rotina2.tempoMs);
        
    }
}
